Updates on installation from packages. (#1574)
plone/documentation: docs/install/install-from-packages.md

The hardware requirements below give a rough estimate of the minimum hardware setup needed for a Plone server.

A single Plone installation is able to run many Plone sites.

- Installation of the Plone backend and Classic UI frontend requires a minimum of 256 MB of RAM and 2GB of disk swap space.
- Installation of the Volto frontend requires a minimum of 2GB of RAM.
- After installation, running Plone requires a minimum of 256 MB RAM and 512 MB of disk swap space per Plone site.
    2 GB or more RAM per Plone site is recommended.
- Minimum 512 MB hard disk space is required.
    40 GB or more hard disk space is recommended.

> - `Add-on` products and caching solutions may also increase RAM and disk swap space requirements.

Reviewer: In the above section:

- Please revert dedents.
- Please revert the replacement of the warning with the bullet point. See this discussion https://github.com/plone/documentation/pull/1561#discussion_r1375325340. I think it is necessary to communicate this information at installation to prevent a lot of frustration.

    If you get Plone to run on Windows alone, please feel free to document and share your process.
    ```

> -   [Python](https://www.python.org/downloads/) 3.8, 3.9, 3.10 or 3.11 at the time of writing. Verify with the [Release notes](https://plone.org/download/releases/).

Reviewer: This makes us look careless. Maintenance of this should be part of releasing Plone. @mauritsvanrees do you have a checklist of things that need to get done in documentation upon a release?

Instead we could define a string of supported Python versions in `conf.py` as we do with other substitutions: https://github.com/plone/documentation/blob/6.0/docs/conf.py#L323-L327. I think we should, because we also use "Python >= 3.8" and similar incorrect version support (we don't support 3.12).

```suggestion
-   [Python](https://www.python.org/downloads/) 3.8, 3.9, 3.10, or 3.11.
```

Author: I define a substitution var in conf.py.

> -   {pipx}`pipx`

Reviewer: ```suggestion
-   {term}`pipx`
```

> -   {term}`Cookiecutter`

Reviewer: Let's just delete it as cookiecutter is not a pre-requisite to install Plone 6, now with pipx in use.

```suggestion
```

-   {term}`nvm`

> -   {term}`Node.js` LTS

Reviewer: We need to make a distinction from LTS 20.x which does not yet work with Volto.

```suggestion
-   {term}`Node.js` LTS 18.x
```

-   {term}`Node.js` LTS 16.x (see "Update" note)
-   {term}`Node.js` LTS
-   {term}`Yeoman`
-   {term}`Yarn`
-   {term}`GNU make`

> -   {term}`Docker`

Reviewer: Why delete Docker from requirements? It's still required.

Author: I bring back Docker to the pre-requisites and postpone the removal to the point in time that Docker is optional. Which is wanted because this chapter is about installing Plone for development without Docker at all.

    If you get Plone to run on Windows alone, please feel free to document and share your process.
    ```

> -   `Python`

Reviewer: ```suggestion
-   Python {SUPPORTED_PYTHON_VERSIONS}
```

-   {term}`pipx`
-   {term}`nvm`

> -   {term}`Node.js` LTS 18.x

Reviewer 2: in fact, its' 20.x (even for 17 series)

Reviewer 2: ```suggestion
-   {term}`Node.js` LTS 20.x
```

Reviewer: But, wait. We recommend 18.x LTS: https://github.com/plone/volto#supported-nodejs-versions.

Reviewer: We also do not mention adding support for Node.js LTS 20.x in the Volto release notes or upgrade guide, at least I could not find any mention of it.

-   {term}`Yeoman`
-   {term}`Yarn`
-   {term}`GNU make`
-   {term}`Docker`

(install-prerequisites-python-label)=

#### Python

Installing Python is beyond the scope of this documentation.
However, it is recommended to use a Python version manager, {term}`pyenv` that allows you to install multiple versions of Python on your development environment without destroying your system's Python.

> Plone supports the following Python versions: {SUPPORTED_PYTHON_VERSIONS}.

Reviewer: Let's switch to the "or" form.

```suggestion
Plone requires Python version {SUPPORTED_PYTHON_VERSIONS}.
```

Installing Python is beyond the scope of this documentation.
However, it is recommended to use a Python version manager, [`pyenv`](https://github.com/pyenv/pyenv) that allows you to install multiple versions of Python on your development environment without destroying your system's Python.

> Install {pipx}`pipx`.

Reviewer: ```suggestion

(install-prerequisites-pipx-label)=

#### pipx

Install {term}`pipx`.
```

```shell
pip install pipx
```

(install-prerequisites-cookiecutter-label)=

> #### Cookiecutter

Reviewer: Let's just delete it as cookiecutter is not a pre-requisite to install Plone 6, now with pipx in use.

```suggestion
```

> Installation and upgrade is done below in {ref}`install-packages-install-label`.

Reviewer: Let's just delete it as cookiecutter is not a pre-requisite to install Plone 6, now with pipx in use.

```suggestion
```

    This command also activates that version.

    ```shell

>     nvm install --lts

Reviewer: That will install v20. Let's use this:

```suggestion
    nvm install "lts/*"
    nvm use 18
```

Also line 127 should be:

```
Install or update the supported LTS versions of Node.js, then activate the version supported in Volto.
```

And delete line 128.

Reviewer: This got missed.

>     This command also activates that version.

Reviewer: The above two lines should be combined into one line.

```suggestion
1.  Install or update the supported LTS versions of Node.js, then activate the version supported in Volto.
```

    ```shell
    nvm install "lts/*"

>     nvm use 18

Reviewer 2: ```suggestion
    nvm use 20
```

(install-packages-install-label)=

## Install Plone 6

> We install Plone 6 with {term}`pip` or {term}`pipx`, {term}`Cookiecutter`, {term}`mxdev`, {term}`make`, and other developer tools.

Reviewer: Do any cookiecutter steps actually run `pip` anymore?

```suggestion
We install Plone 6 with {term}`pipx`, {term}`Cookiecutter`, {term}`mxdev`, {term}`make`, and other developer tools.
```

(install-packages-install-label)=

#### Install Docker

Install [Docker Desktop](https://docs.docker.com/get-docker/) for your operating system.

Docker Desktop includes all Docker tools.

> 

Reviewer: Please restore this requirement.

```suggestion

#### Install Docker

Install [Docker Desktop](https://docs.docker.com/get-docker/) for your operating system.

Docker Desktop includes all Docker tools.

```

```

> Run `cookiecutter` to create a Plone project skeleton using the Cookiecutter {term}`cookiecutter-plone-starter` with the following command.

Reviewer: ```suggestion
Issue the following command to install or update `cookiecutter`, then run it to create a Plone project skeleton using the Cookiecutter {term}`cookiecutter-plone-starter`.
```

> This also updates cookiecutter.

Reviewer: ```suggestion
```

cd my_project
```

Issue the following command to install or update `cookiecutter`, then run it to create a Plone project skeleton using the Cookiecutter {term}`cookiecutter-plone-starter`.

```shell
pipx run cookiecutter gh:collective/cookiecutter-plone-starter
```

You will be presented with a series of prompts.
You can accept the default values in square brackets (`[default-option]`) by hitting the {kbd}`Enter` key, or enter your preferred values.
For ease of documentation, we will use the default values.

```console
% pipx run cookiecutter gh:collective/cookiecutter-plone-starter

Cookiecutter Plone Starter 
================================================================================

Sanity checks
--------------------------------------------------------------------------------
  [1/5] Python: ✓
  [2/5] Node: ✓
  [3/5] yo: ✓
  [4/5] Docker: ✓
  [5/5] git: ✓

Project details
--------------------------------------------------------------------------------

>   [1/19] Project Title (Project Title): Plone Conference Website 2070

Reviewer: LOL we'll never have to update it again during our lifetimes.

  [2/19] Project Description (A new project using Plone 6.): 
  [3/19] Project Slug (Used for repository id) (plone-conference-website-2070): 
  [4/19] Project URL (without protocol) (plone-conference-website-2070.example.com): 
  [5/19] Author (Plone Foundation): Elli
  [6/19] Author E-mail (collective@plone.org): elli@plone.org
  [7/19] Python Package Name (plone_conference_website_2070): 
  [8/19] Volto Addon Name (volto-plone-conference-website-2070): 
  [9/19] Choose a Python Test Framework
    1 - pytest
    2 - unittest
    Choose from [1/2] (1): 
  [10/19] Plone Version (6.0.8): 
  [11/19] Should we use Volto Alpha Versions? (No): yes

>   [12/19] Volto Version (18.0.0-alpha.1): 

Reviewer: Should we choose the installation of alphas or enter a stable version? 

```suggestion
  [12/19] Volto Version (18.0.0-alpha.1): 
```

Author: I chose explicitly the alpha, because this, from package, is in most cases setting up Plone for development, not for production.

This will take a few minutes.
☕️
First the backend, then the frontend will be installed.

> When the process completes successfully, it will exit with no message.

Reviewer: Please restore the above section. See: https://github.com/collective/cookiecutter-plone-starter/blob/138b1391a9612a9b179b711a9e5f1e6b1ed2af85/%7B%7B%20cookiecutter.project_slug%20%7D%7D/frontend/Makefile.default#L58-L59

```suggestion

If mrs-developer is already installed, when the process completes successfully, it will exit with no message.
```

This will take a few minutes.
☕️
First the backend, then the frontend will be installed.

> If mrs-developer is already installed, when the process completes successfully, it will exit with no message.

Reviewer: I have not verified whether installing mrs-developer exits without a message. It used to exit with `✨  Done in 98.97s.`, but haven't tried it in a while. I'll go with what you say, "no message". Thus we can delete this line, as it is redundant to line 363 below.

Author: I took a look how and when mrs-developer is installed. It seems that nowadays mrs-developer is installed without a dialog if it is needed, which is the case if  a configuration file `mrs.developer.json` is present. 
So I'll kick out the lines about the mrs-developer dialog.

See Makefile: `npx -p mrs-developer`
```
.PHONY: develop
develop: ## Runs missdev in the local project (mrs.developer.json should be present)
	if [ -f $$(pwd)/tsconfig.json ]; then npx -p mrs-developer missdev --config=tsconfig.json --output=addons --fetch-https; else  npx -p mrs-developer missdev --config=jsconfig.json --output=addons --fetch-https;fi

```
